buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ForbiddenPatternsTask.java

```diff
@@ -121,13 +115,19 @@ public void checkInvalidPatterns() throws IOException {
             throw new GradleException("Found invalid patterns:\n" + String.join("\n", failures));
         }
 
+        File outputMarker = getOutputMarker();
         outputMarker.getParentFile().mkdirs();
         Files.write(outputMarker.toPath(), "done".getBytes(StandardCharsets.UTF_8));
     }
 
     @OutputFile
     public File getOutputMarker() {
-        return outputMarker;
+        return new File(getProject().getBuildDir(), "markers/" + getName());
+    }
+
+    @Input
+    public Map<String, String> getPatterns() {
+        return patterns;
```

Comment:
We should wrap this with `Collections.unmodifiableMap` to prevent users from circumventing the `rule` method.